### PR TITLE
run sanity checks commands from an empty temporary directory (rather than the software install directory)

### LINKED ISSUE #4044: `--sanity-check-only` runs commands in installation dir
This leads to problems if the commands have output files since these will get placed in the installation

### PR DESCRIPTION
**Pull request by lexming**
Fixes #4044

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -3739,16 +3739,16 @@ def xs2str(xs):
         if self.toolchain.mpi_family() and self.toolchain.mpi_family() in toolchain.OPENMPI:
             env.setvar('OMPI_MCA_rmaps_base_oversubscribe', '1')
 
-        # change to install directory (better environment for running tests)
-        if os.path.isdir(self.installdir):
-            change_dir(self.installdir)
+        # run sanity checks from an empty temp directory
+        # using the build or installation directory can produce false positives and polute them with files
+        sanity_work_dir = tempfile.mkdtemp(prefix='eb-saniy-check-')
```
Comment (boegel) [inline easybuild/framework/easyblock.py]: @lexming Are you testing whether or not I'll catch the typo here? ;)

Comment (lexming) [reply easybuild/framework/easyblock.py]: ohh good one :smile:

Comment (boegel) [reply easybuild/framework/easyblock.py]: fixed in 2389fc6

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**